src/common/version-to-skip.helpers.ts

```diff
@@ -0,0 +1,125 @@
+import * as vscode from 'vscode'
+
+import {
+  ANG_CLI,
+  ANG_CORE,
+  IVersionStatus,
+  checkForUpdate,
+} from '../file/package-manager'
+
+import { UpgradeChannel } from './enums'
+import { getUpgradeChannel } from './upgrade-channel.helpers'
+
+export const VERSION_TO_SKIP_KEY = 'ng-evergreen.versionToSkip'
+
+export function getVersionToSkip(): string | undefined {
+  return vscode.workspace.getConfiguration().get(VERSION_TO_SKIP_KEY)
+}
+
+export function versionToSkipExists(): boolean {
+  const upgradeVersion = getVersionToSkip()
+  return !!upgradeVersion && upgradeVersion !== ''
+}
+
+export function storeVersionToSkip(versionToSkip: string): void {
+  vscode.workspace.getConfiguration().update(VERSION_TO_SKIP_KEY, versionToSkip)
+}
+
+export function clearVersionToSkip(): void {
+  vscode.workspace.getConfiguration().update(VERSION_TO_SKIP_KEY, undefined)
+}
+
+export async function shouldUpgradeToNewVersion(): Promise<boolean> {
+  const upgradeChannel = getUpgradeChannel()
+  const versionToSkip = getVersionToSkip()
+  const ngCoreVersion = await checkForUpdate(ANG_CORE)
```

Comment:
@bjsawyer we need to refactor the calls to 
```
  const coreOutdated = await checkForUpdate(ANG_CORE)
  const cliOutdated = await checkForUpdate(ANG_CLI)
```
or in general the `shouldUpdateToNewVersion` logic, so that there’s only one such function. The current logic is faulty and due to be fixed by an upcoming ticket, if we can get this refactor done now, it’ll be an easy change

Reply:
@duluca which upcoming ticket are you referring to? Not totally clear on the purpose of the refactor.

Reply:
See #35 for details